src/historywork/GetRemoteFileWork.cpp

```diff
@@ -30,19 +31,36 @@ GetRemoteFileWork::~GetRemoteFileWork()
 void
 GetRemoteFileWork::getCommand(std::string& cmdLine, std::string& outFile)
 {
-    auto archive = mArchive;
-    if (!archive)
+    mCurrentArchive = mArchive;
+    if (!mCurrentArchive)
     {
-        archive = mApp.getHistoryManager().selectRandomReadableHistoryArchive();
+        mCurrentArchive =
+            mApp.getHistoryManager().selectRandomReadableHistoryArchive();
     }
-    assert(archive);
-    assert(archive->hasGetCmd());
-    cmdLine = archive->getFileCmd(mRemote, mLocal);
+    assert(mCurrentArchive);
+    assert(mCurrentArchive->hasGetCmd());
+    cmdLine = mCurrentArchive->getFileCmd(mRemote, mLocal);
 }
 
 void
 GetRemoteFileWork::onReset()
 {
     std::remove(mLocal.c_str());
 }
+
+Work::State
+GetRemoteFileWork::onSuccess()
+{
+    mApp.getHistoryArchiveMonitor().markOperationSucceeded(
```

Comment:
pattern is a bit weird: why not have those metrics attached directly to the `HistoryArchive` object instead of having to expose the ownership to the `*Work` classes? It seems that they are important enough to be first classes into that object.

Then I can just call `mCurrentArchive.recordFailure()`.

Reply:
I made this only because HistoryArchive classes live in Config. I do not think that having mutable objects inside Config is a good idea. I'll try to find a better place for HistoryArchive objects and then make them mutable.

Reply:
yeah what we could do is add a new class that represents the history archive instance (that deals with invocation of archive related work, monitoring, etc), that way we start to formalize the interface with archive a little better (later we can implement proper "load balancing" between archives if we have an interface) and maybe rename the existing HistoryArchive as HistoryArchiveConfiguration